applications/schedview-snapshot/secrets.yaml

```diff
@@ -0,0 +1,9 @@
+# Follow the example of plot-navigator, but only getting the aws-credentials
+
+"aws-credentials.ini":
+  description: >-
+    Cloud storage for access to the LFA, from which scheduler snapshots
+    can be loaded.
+  copy:
+    application: nublado
+    key: "aws-credentials.ini"
```

Comment:
Does this application need all of the credentials that a user would have, to all of the S3 buckets, or does it only need access to a specific subset of S3 buckets? I'm worried that this is too many permissions and it would be better to create more scoped credentials specific to the application, but I'm not sure what the application does. If it should be able to access anything a user can access from a notebook (read and write), then this is correct.

Reply:
The application needs read access to the LFA (is this just one bucket, or many?): all of URIs for objects I need access to begin `s3://rubin:rubinobs-lfa-cp/Scheduler`. (There will need to be a deployment at the summit as well, with access to the corresponding summit URIs, but I'm not there yet, and summit staff will probably be the ones to configure phalanx for that environment, and it will be handled in a separate ticket and pull request.)
I don't know how to get credentials that limit the access to this subset. When I asked on slack (see the thread I linked to in the original comment), it sounded like trying this would make things more complicated, and it would be acceptable and simpler to use the nublado credentials. If I can get help getting something more restricted, that would be great.

Reply:
Well, my security advice would be to get a more restricted token that only has read access to the objects or bucket that you are using, put that token in the value for the `aws-credentials.ini` secret in the Vault path for this application, and remove the copy directive (eveything else about this `secrets.yaml` is file). Using correctly-scoped tokens is generally better, in part because it will protect against mistakes.

But I'm not involved in running the USDF cluster and it's their security decision, so if they really want you to use the more-powerful tokens that are injected into user notebooks, it's their call. @ktlim, are you the person who handles this?

Reply:
The rubin-summit-users credential is currently used for read-only access to all buckets that staff users will need access to on an everyday basis so that multiple profiles don't need to be used.  (The credential also has read/write access to the buckets containing Butler repo user collections.)

In the near future (within the next few weeks) when we move to the embargo rack, which necessitates multiple profiles anyway due to having multiple endpoint URLs, I have been considering having an additional LFA-specific read-only credential that would be used with `s3://lfa@rubinobs-lfa-cp/Scheduler` URIs.  That profile would also be installed in the nublado `butler-secret`, but a more limited custom LFA-only profile could also be used.

Reply:
It sounds like it'll be a couple of weeks before things are ready to switch to more limited secrets, so this copy directive is correct for now. But you'll still need to change the reference to butler-secret to schedview-snapshot instead, down below where you create the `VaultSecret` resource, since we're switching everything over to per-app secrets. usdfint is done, and usdfdev and maybe even usdfprod should hopefully be done next Thursday.

Once the more restrictive secrets have been implemented, the more restricted secret should be the only one included in the schedview-snapshot secret, and at that point the copy directive here should be removed.